src/js/fc.js

```diff
@@ -815,4 +815,155 @@ const FC = {
         }
         return versionPidDefaults;
     },
+
+    // these values are manufactured to unlock all the functionality of the configurator, they dont represent actual hardware
+    generateVirtualConfig() {
```

Comment:
I think maybe is a good idea to move to a different file, like `VirtualFc.js` or similar, if the code is clean, all the code that references this virtual thing. I think it will be easier to maintain. Others thoughts about this?

Reply:
I can move it to another file, I know it isn't the prettiest solution but this is the only way to get around editing every tab to support this feature. Thanks to this only the OSD tab needs extra code to function.

Reply:
True. But wait to see what others think. It has sense to be here, with all the FC config, but maybe is better in another file to help to maintain if we can move there all the code that need changes when the Betaflight firmware updates to this file.

Reply:
The current solution is more transparent and populating existing `FC` variable. Having to change every tab to adjust to a `VIRTUALFC` needs more adjustment in the code. Imho like to keep it centralized.

Reply:
I haven't looked into this in detail yet, but duplicating structures sounds like a maintenance nightmare.
One option that JavaScript has to allows separation of functionality is monkey patching - maybe this could lend itself for the situation at hand?

Reply:
I'm not sure what you mean by duplicating structures, `generateVirtualConfig` fills the FC object with values that unlock the functionality of every tab without additional modification, this only needs minimal maintenance when a new feature is added to a tab. There is always the option to break this function up into logical parts or use monkey patching somehow, but I think that would only make this simple function more convoluted and harder to maintain.

Reply:
@visdauas: I was referring to 

> Having to change every tab to adjust to a `VIRTUALFC` needs more adjustment in the code.

Reply:
I think this only needs rebase and squash commits into 1 before being accepted?

Reply:
I still agree with @McGiverGim here: For now this is ok, but once we need to support multiple firmware versions this will get bigger and messier (especially once the OSD part is in here as well), and we should be thinking about a way to structure this in a better way.